### PR TITLE
Feature feedback in app component

### PR DESCRIPTION
**Pull request by Dzavala-dev**
## Useful links
- [Contributing](https://github.com/mercadolibre/fury_andesui-android/blob/develop/CONTRIBUTING.md) has more information and tips for a great pull request. This will give you a nice introduction about how to contribute with Andes.
- [Wiki](https://github.com/mercadolibre/fury_andesui-android/wiki) has all the info you need to start with Andes. Even I heard there's a video there...

## Motivation
Tell us why you are here and why you took time to develop new code for our beloved lib.

Im helping with create a new component called "Feedback In App"

## Description
Please describe your intentions with this new code you want to introduce inside Andes UI lib.
Try to answer those questions:
- What is it about? It's a component thats just redirect the user to a CB
- How does it work? When a user need to tell us or give us point about the item or anything on the platform just need to click on "ayudanos a mejorar" 

## Affected component
Is highly probable that this new code directly affects one (or more?) components inside this lib. Tell us who they are!

## Frontify component link
The UX team behind Andes keeps all the Andes Components inside the Frontify site. Each component has its own section. Let us know the section of the component you want to modify.

## Screenshots
Captura de Pantalla 2020-12-28 a la(s) 11 21 43

## Checks
Are you sure you followed all those steps? In such case, let's say with me "I solemnly declare that ...":
- [X] I have coded an example inside the Demo App,
- [ ] I have added proper tests,
- [ ] I have modified the [Changelog](https://github.com/mercadolibre/fury_andesui-android/blob/develop/CHANGELOG.md) file,
- [X] I have updated GitHub wiki,
- [ ] I have the explicit approval of one or more members of the UX Team,
- [ ] I have updated the version in gradle.properties file.

## Change type
This is easy, let us know what this code is about:
- [X] This code adds a new component
- [ ] This code includes improvements to an existent component
- [ ] This code improves or modifies ONLY the Demo App.

## Check common errors
Whether you are an Android or iOS developer and if you're still there then we'll give you a present:
https://proandroiddev.com/how-to-maximize-androids-ui-reusability-5-common-mistakes-cb2571216a9f

### CODE REVIEW

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+import com.mercadolibre.android.andesui.R
+import com.mercadolibre.android.andesui.button.AndesButton
+import com.mercadolibre.android.andesui.feedbackinapp.align.AndesFeedbackInAppAlign
+import com.mercadolibre.android.andesui.feedbackinapp.factory.FeedbackInAppAttrs
+import com.mercadolibre.android.andesui.feedbackinapp.factory.FeedbackInAppAttrsParser
+import com.mercadolibre.android.andesui.feedbackinapp.factory.FeedbackInAppConfiguration
+import com.mercadolibre.android.andesui.feedbackinapp.factory.FeedbackInAppConfigurationFactory
+
+
+@Suppress("TooManyFunctions")
```
Comment (mauriciogigena) [inline components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt:19]: si no te marca este warning no hace falta que agregues el suppress

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        setupComponents()
+    }
+
+    /**
+     * Responsible for setting up all properties of each component that is part of this message.
+     * Is like a choreographer ;)
+     *
+     */
+    private fun setupComponents() {
+
```
Comment (mauriciogigena) [inline components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt:105]: aca te quedo un enter de mas

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        if (id == NO_ID) { // If this view has no id
+            id = View.generateViewId()
+        }
+    }
+
+    /**
+     * Gets data from the config and sets to the text component of this button.
+     *
+     */
+    private fun setupBodyComponent(config: FeedbackInAppConfiguration) {
```
Comment (mauriciogigena) [inline components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt:139]: el metodo deberia referirse al text o message, no a body

Comment (mauriciogigena) [reply components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt:139]: ademas te quedo un salto de linea

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+        feedbackinappMessage.text = config.feedbackInAppMessageText
+
+    }
+
+    fun setupLinkAction(config: FeedbackInAppConfiguration) {
+
```
Comment (mauriciogigena) [inline components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt:146]: saltos de linea

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/factory/FeedbackInAppAttrsParser.kt [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        val deeplink: String?,
+        val buttonLabel: String?
+)
+
+/**
+ * This object parse the attribute set and return an instance of AndesButtonAttrs to be used by AndesButton
+ */
+internal object FeedbackInAppAttrsParser {
+
+    private const val ANDES_FeedbackInApp_HIERARCHY_LOUD = "100"
```
Comment (mauriciogigena) [inline components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/factory/FeedbackInAppAttrsParser.kt:27]: estos lo usamos en el componente feeback in app? no tiene hierarchy

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/factory/FeedbackInAppConfigurationFactory.kt [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+)
+
+@Suppress("TooManyFunctions")
+internal object FeedbackInAppConfigurationFactory {
+
+    fun create(context: Context, andesFeedbackInAppAttrs: FeedbackInAppAttrs): FeedbackInAppConfiguration {
+
+        return with(andesFeedbackInAppAttrs) {
+            FeedbackInAppConfiguration(
+
```
Comment (mauriciogigena) [inline components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/factory/FeedbackInAppConfigurationFactory.kt:24]: salto de linea

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/hierarchy/AndesFeedbackInAppHierarchy.kt [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+import com.mercadolibre.android.andesui.feedbackinapp.AndesFeedbackInApp
+
+/**
+ * Utility class that does two things: Defines the possible hierarchies an [AndesFeedbackInApp] can take because it's an enum, as you can see.
+ * But as a bonus it gives you the proper implementation so you don't have to make any mapping.
+ *
+ *
+ * @property hierarchy Possible hierarchies that an [AndesFeedbackInApp] may take.
+ */
+enum class AndesFeedbackInAppHierarchy {
```
Comment (mauriciogigena) [inline components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/hierarchy/AndesFeedbackInAppHierarchy.kt:12]: no tiene hierarchies el componente

File: components/src/main/res/layout/andes_layout_feedbackinapp.xml [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+            android:visibility="visible"
+            app:layout_constraintBottom_toTopOf="@+id/andes_feedbackinapp_message" />
+
+        <TextView
+            android:id="@+id/andes_feedbackinapp_message"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginBottom="17dp"
+            android:gravity="center_horizontal"
+            android:textColor="#73000000"
```
Comment (mauriciogigena) [inline components/src/main/res/layout/andes_layout_feedbackinapp.xml:32]: esto deberia estr en colors file

File: components/src/main/res/layout/andes_layout_feedbackinapp.xml [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+            android:textIsSelectable="false"
+            android:visibility="visible"
+            app:layout_constraintBottom_toTopOf="@+id/andes_feedback_button"
+            app:layout_constraintTop_toTopOf="@id/andes_feedback_divider" />
+
+        <com.mercadolibre.android.andesui.button.AndesButton
+            android:id="@+id/andes_feedback_button"
+            android:layout_width="match_parent"
+            android:layout_height="32dp"
```
Comment (mauriciogigena) [inline components/src/main/res/layout/andes_layout_feedbackinapp.xml:41]: las medidas en dimens

**Review by mauriciogigena — CHANGES_REQUESTED** [review]

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt (hunk 1 above)
Comment (Dzavala-dev) [reply components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt:19]: Remuevo todos los "@Suppress"??

Comment (mauriciogigena) [reply components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt:19]: solo este por ahora

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/hierarchy/AndesFeedbackInAppHierarchy.kt (hunk 7 above)
Comment (Dzavala-dev) [reply components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/hierarchy/AndesFeedbackInAppHierarchy.kt:12]: Listo ya se borro el Hierarchy

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/factory/FeedbackInAppAttrsParser.kt (hunk 5 above)
Comment (Dzavala-dev) [reply components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/factory/FeedbackInAppAttrsParser.kt:27]: Listo ya se borro el Hierarchy

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt (hunk 3 above)
Comment (Dzavala-dev) [reply components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt:139]: ya se modigico a setupMessageComponent

File: components/src/main/res/layout/andes_layout_feedbackinapp.xml (hunk 9 above)
Comment (Dzavala-dev) [reply components/src/main/res/layout/andes_layout_feedbackinapp.xml:41]: Listo ya se agregaron 
Captura de Pantalla 2020-12-28 a la(s) 12 34 11

File: components/src/main/res/layout/andes_layout_feedbackinapp.xml (hunk 8 above)
Comment (Dzavala-dev) [reply components/src/main/res/layout/andes_layout_feedbackinapp.xml:32]: Listo

Captura de Pantalla 2020-12-28 a la(s) 12 34 47

File: components/build.gradle [hunk 10]
```diff
@@ -76,4 +76,5 @@ dependencies {
     testImplementation test.assertj
     testImplementation test.robolectric
     testImplementation test.mockitoInline
+    implementation 'androidx.constraintlayout:constraintlayout:2.0.4'
```
Comment (mauriciogigena) [inline components/build.gradle:79]: por que se importa esto?

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt (hunk 1 above)
Comment (mauriciogigena) [reply components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt:24]: este quedo, habia que sacarlo

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt [hunk 11]
```diff
@@ -0,0 +1,226 @@
...
+     */
+    var cardBackgroundColor: String?
+        get() = andesFeedbackInAppAttrs.cardBackgroundColor
+        set(value) {
+            andesFeedbackInAppAttrs = andesFeedbackInAppAttrs.copy(cardBackgroundColor = value)
+           setupcardBackground(createConfig())
+        }
+
+    private lateinit var feedbackContainer: ConstraintLayout
+    private lateinit var feedbackdivider: View
```
Comment (mauriciogigena) [inline components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt:77]: Divider, con D

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt [hunk 12]
```diff
@@ -0,0 +1,226 @@
...
+        }
+
+    /**
+     * Getter and setter for [cardBackgroundColor].
+     */
+    var cardBackgroundColor: String?
+        get() = andesFeedbackInAppAttrs.cardBackgroundColor
+        set(value) {
+            andesFeedbackInAppAttrs = andesFeedbackInAppAttrs.copy(cardBackgroundColor = value)
+           setupcardBackground(createConfig())
```
Comment (mauriciogigena) [inline components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt:73]: Card con C

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt [hunk 13]
```diff
@@ -0,0 +1,226 @@
...
+        throw IllegalStateException(
+                "Constructor without parameters in Andes Message is not allowed. You must provide some attributes."
+        )
+    }
+
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
+        initAttrs(attrs)
+    }
+
+    @Suppress("unused", "LongParameterList")
```
Comment (mauriciogigena) [inline components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt:93]: Este no deberia hacer falta, borrarlo y fijate si te marca algo

File: components/src/main/res/layout/andes_layout_feedbackinapp.xml [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+        android:layout_height="130dp"
+        android:animateLayoutChanges="true"
+        android:visibility="visible">
+
+        <View
+            android:id="@+id/andes_feedback_divider"
+            android:layout_width="match_parent"
+            android:layout_height="@dimen/andes_feedbackinapp_divider_height"
+            android:layout_margin="@dimen/andes_feedback_divider_margin"
+            android:layout_marginTop="15dp"
```
Comment (mzangl-meli) [inline components/src/main/res/layout/andes_layout_feedbackinapp.xml:19]: use dimen

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt [hunk 15]
```diff
@@ -0,0 +1,226 @@
...
+    ) : super(context) {
+        initAttrs(cardText, showTopDivider, cardTextAlignment, backgroundColor, buttonLabel)
+    }
+
+    private fun initAttrs(attrs: AttributeSet?) {
+        andesFeedbackInAppAttrs = FeedbackInAppAttrsParser.parse(context, attrs)
+        setupComponents()
+    }
+
+    @Suppress("LongParameterList")
```
Comment (mauriciogigena) [inline components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt:110]: Este no deberia hacer falta, borrarlo y fijate si te marca algo

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt [hunk 16]
```diff
@@ -0,0 +1,226 @@
...
+     *
+     */
+    private fun initComponents() {
+        val container = LayoutInflater.from(context).inflate(R.layout.andes_layout_feedbackinapp,
+                this, true)
+        feedbackContainer = container.findViewById(R.id.andes_feedback_container)
+        feedbackContainer.background
+        feedbackdivider = container.findViewById(R.id.andes_feedback_divider)
+        feedbackdivider.visibility = View.GONE
+        feedbackinappMessage = container.findViewById(R.id.andes_descriptionText)
```
Comment (mauriciogigena) [inline components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt:146]: Aca todo se viene llamando "andes_feedback_algo y este es andes_descriptionText, incluso esta escrito de manera diferente

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt [hunk 17]
```diff
@@ -0,0 +1,226 @@
...
+                    throw IllegalStateException("Cannot initialize a link action")
+                else ->
+                    Log.d("AndesMessage", "Cannot initialize a link action")
+            }
+        }
+    }
+
+    private fun setupShowtopdivider(config: FeedbackInAppConfiguration) {
+        if (config.showTopDivider) {
+            feedbackdivider.visibility = View.VISIBLE
```
Comment (mauriciogigena) [inline components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt:198]: no es un error, pero View deberia estar en los imports

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt [hunk 18]
```diff
@@ -0,0 +1,226 @@
...
+        } else {
+            feedbackdivider.visibility = View.GONE
+        }
+    }
+
+    private fun setupcardBackground(config: FeedbackInAppConfiguration) {
+        if (config.cardBackgroundColor.isNullOrEmpty()) {
+            this.setBackgroundColor(ContextCompat.getColor(context, R.color.andes_bg_color_white))
```
Comment (mauriciogigena) [inline components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt:206]: el color por defecto deberia ser un gray

File: components/src/main/res/layout/andes_layout_feedbackinapp.xml [hunk 19]
```diff
@@ -0,0 +1,57 @@
...
+            android:layout_marginBottom="@dimen/andes_feedback_divider_margin"
+            android:background="@color/andes_gray_450_solid"
+            android:visibility="visible"
+            app:layout_constraintBottom_toTopOf="@+id/andes_descriptionText"
+            app:layout_constraintTop_toTopOf="parent"
+            tools:layout_editor_absoluteX="16dp" />
```
Comment (mauriciogigena) [inline components/src/main/res/layout/andes_layout_feedbackinapp.xml:25]: esto deberia estar en dimens

File: components/src/main/res/layout/andes_layout_feedbackinapp.xml (hunk 14 above)
Comment (mauriciogigena) [inline components/src/main/res/layout/andes_layout_feedbackinapp.xml:19]: dimens, y ademas me parece que esta mal el numero, siempre usamos pares, revisaloo en zeplin

File: components/src/main/res/values/dimens_feedbackinapp.xml [hunk 20]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+    <!--Description Text-->
+    <dimen name="andes_feedback_textmarginStart">50dp</dimen>
+    <dimen name="andes_feedback_textmarginBottom">17dp</dimen>
```
Comment (mauriciogigena) [inline components/src/main/res/values/dimens_feedbackinapp.xml:6]: revisa estos valores con zeplin

**Review by mauriciogigena — CHANGES_REQUESTED** [review]

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt (hunk 1 above)
Comment (Dzavala-dev) [reply components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt:24]: Si lo saque, pero me pone warning y no pasa el PR , por eso lo volvi a poner ayer @mauriciogigena

File: components/build.gradle (hunk 10 above)
Comment (mdobratinich) [reply components/build.gradle]: Si hay que sacar ese import

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt (hunk 12 above)
Comment (mdobratinich) [reply components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt:73]: +1

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt [hunk 21]
```diff
@@ -0,0 +1,226 @@
...
+    }
+
+    @Suppress("LongParameterList")
+    private fun initAttrs(
+            cardText: String,
+            showTopDivider: Boolean,
+            cardTextAlignment: AndesFeedbackInAppAlign,
+            backgroundColor: String,
+            buttonLabel: String?
+
```
Comment (mdobratinich) [inline components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt:110]: Esta linea esta de mas

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt [hunk 22]
```diff
@@ -0,0 +1,226 @@
...
+            id = View.generateViewId()
+        }
+    }
+
+    /**
+     * Gets data from the config and sets to the text component of this button.
+     *
+     */
+    private fun setupMessageComponent(config: FeedbackInAppConfiguration) {
+        if (config.cardText == null || config.cardText.isEmpty()) {
```
Comment (mdobratinich) [inline components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt:159]: Queda mas prolijo usar:
`config.cardText.isNullOrEmpty()`

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt [hunk 23]
```diff
@@ -0,0 +1,226 @@
...
+            feedbackinappMessage.visibility = View.GONE
+        } else {
+            feedbackinappMessage.visibility = View.VISIBLE
+            feedbackinappMessage.text = config.cardText
+        }
+    }
+
+    private fun setupLabelComponent(config: FeedbackInAppConfiguration) =
+            if (config.action == null || config.action.isEmpty()) {
```
Comment (mdobratinich) [inline components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt:168]: idem anterior

File: components/src/main/res/layout/andes_layout_feedbackinapp.xml [hunk 24]
```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="utf-8"?>
+<merge xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    tools:parentTag="androidx.cardview.widget.CardView">
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:id="@+id/andes_feedback_container"
+        android:layout_width="match_parent"
+        android:layout_height="130dp"
```
Comment (mdobratinich) [inline components/src/main/res/layout/andes_layout_feedbackinapp.xml:10]: mover a dimen

File: components/src/test/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInAppTest.kt [hunk 25]
```diff
@@ -0,0 +1,81 @@
...
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [Build.VERSION_CODES.LOLLIPOP])
+class AndesFeedbackInAppTest {
+    private lateinit var attrs: FeedbackInAppAttrs
+
+    @Test
+    fun `FeedbackInApp, CENTER, False`() {
+        attrs = FeedbackInAppAttrs(
+                cardText = "Andes FeedbackInApp",
+                buttonLabel = "Ayuda a mejorar",
```
Comment (mdobratinich) [inline components/src/test/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInAppTest.kt:20]: Los string pasalos a constantes o usa los String declarados en el String.xml

File: components/src/test/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInAppTest.kt [hunk 26]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    @Test
+    fun `FeedbackInApp, CENTER, True`() {
+        attrs = FeedbackInAppAttrs(
+                cardText = "Andes FeedbackInApp",
+                buttonLabel = "Ayuda a mejorar",
+                cardTextAlignment = AndesFeedbackInAppAlign.CENTER,
+                showTopDivider = true,
+                cardBackgroundColor = "#FF7733"
```
Comment (mdobratinich) [inline components/src/test/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInAppTest.kt:56]: Usa un AndesColor

File: demoapp/src/main/java/com/mercadolibre/android/andesui/demoapp/feature/FeedbackInAppActivity.kt [hunk 27]
```diff
@@ -0,0 +1,129 @@
+package com.mercadolibre.android.andesui.demoapp.feature
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.*
```
Comment (mdobratinich) [inline demoapp/src/main/java/com/mercadolibre/android/andesui/demoapp/feature/FeedbackInAppActivity.kt:9]: evita usar wildcards

File: demoapp/src/main/res/layout/andesui_feedbackinapp_change.xml [hunk 28]
```diff
@@ -0,0 +1,114 @@
...
+        <androidx.constraintlayout.widget.ConstraintLayout
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:isScrollContainer="true">
+
+            <Switch
+                android:id="@+id/showtopdivider"
+                android:layout_width="390dp"
+                android:layout_height="wrap_content"
+                android:layout_marginTop="15dp"
```
Comment (mdobratinich) [inline demoapp/src/main/res/layout/andesui_feedbackinapp_change.xml]: mover a dimen

File: demoapp/src/main/res/layout/andesui_feedbackinapp_change.xml [hunk 29]
```diff
@@ -0,0 +1,114 @@
...
+        </com.mercadolibre.android.andesui.feedbackinapp.AndesFeedbackInApp>
+
+        <androidx.constraintlayout.widget.ConstraintLayout
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:isScrollContainer="true">
+
+            <Switch
+                android:id="@+id/showtopdivider"
+                android:layout_width="390dp"
```
Comment (mdobratinich) [inline demoapp/src/main/res/layout/andesui_feedbackinapp_change.xml]: mover a dimen

File: demoapp/src/main/res/layout/andesui_feedbackinapp_change.xml [hunk 30]
```diff
@@ -0,0 +1,114 @@
...
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintTop_toTopOf="parent"
+                tools:ignore="MissingConstraints" />
+
+            <TextView
+                android:id="@+id/feedbackinapp_align"
+                android:layout_width="156dp"
+                android:layout_height="21dp"
+                android:layout_marginTop="10dp"
+                android:layout_marginBottom="10dp"
```
Comment (mdobratinich) [inline demoapp/src/main/res/layout/andesui_feedbackinapp_change.xml]: idem

File: demoapp/src/main/res/layout/andesui_feedbackinapp_change.xml [hunk 31]
```diff
@@ -0,0 +1,114 @@
...
+                app:layout_constraintStart_toStartOf="@+id/showtopdivider"
+                app:layout_constraintTop_toBottomOf="@+id/showtopdivider"
+                app:layout_constraintVertical_bias="0.769" />
+
+            <Spinner
+                android:id="@+id/spinnerAlign"
+                android:layout_width="128dp"
+                android:layout_height="23dp"
+                android:layout_marginTop="6dp"
+                android:layout_marginBottom="10dp"
```
Comment (mdobratinich) [inline demoapp/src/main/res/layout/andesui_feedbackinapp_change.xml]: idem

File: demoapp/src/main/res/layout/andesui_feedbackinapp_change.xml [hunk 32]
```diff
@@ -0,0 +1,114 @@
...
+            android:layout_height="wrap_content"
+            android:layout_marginBottom="@dimen/button_margin_vertical_small"
+            android:inputType="text"
+            app:andesTextareaMaxLines="3"
+            app:andesTextareaPlaceholder="@string/change_body" />
+
+        <com.mercadolibre.android.andesui.textfield.AndesTextarea
+            android:id="@+id/background"
+            android:layout_width="match_parent"
+            android:layout_height="37dp"
```
Comment (mdobratinich) [inline demoapp/src/main/res/layout/andesui_feedbackinapp_change.xml]: idem

File: demoapp/src/main/res/layout/andesui_feedbackinapp_change.xml [hunk 33]
```diff
@@ -0,0 +1,114 @@
...
+            android:layout_marginBottom="@dimen/button_margin_vertical_small"
+            android:inputType="text"
+            app:andesTextareaPlaceholder="@string/change_Background" />
+
+        <com.mercadolibre.android.andesui.textfield.AndesTextfield
+            android:id="@+id/action_text"
+            android:layout_width="match_parent"
+            android:layout_height="37dp"
```
Comment (mdobratinich) [inline demoapp/src/main/res/layout/andesui_feedbackinapp_change.xml:93]: idem

**Review by mdobratinich — CHANGES_REQUESTED** [review]

File: components/src/main/res/layout/andes_layout_feedbackinapp.xml [hunk 34]
```diff
@@ -5,52 +5,53 @@
...
             android:visibility="visible"
-            app:layout_constraintBottom_toTopOf="@+id/andes_descriptionText"
-            app:layout_constraintTop_toTopOf="parent"
-            tools:layout_editor_absoluteX="16dp" />
+            app:layout_constraintBottom_toTopOf="@+id/andes_feedbackinapp_cardText"
+            app:layout_constraintTop_toTopOf="parent" />
 
         <TextView
-            android:id="@+id/andes_descriptionText"
+            android:id="@+id/andes_feedbackinapp_cardText"
```
Comment (mdobratinich) [inline components/src/main/res/layout/andes_layout_feedbackinapp.xml:26]: Los ids no deben tener camel case... andes_feedbackinapp_card_text

File: components/src/main/res/layout/andes_layout_feedbackinapp.xml [hunk 35]
```diff
@@ -5,52 +5,53 @@
...
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
             android:layout_marginStart="@dimen/andes_feedback_textmarginStart"
+            android:layout_marginTop="@dimen/andes_feedback_divider_margin"
             android:layout_marginEnd="@dimen/andes_feedback_textmarginEnd"
+            android:layout_marginBottom="@dimen/andes_feedback_textmarginBottom"
```
Comment (mdobratinich) [inline components/src/main/res/layout/andes_layout_feedbackinapp.xml:32]: igual que el comentario anterior

File: components/src/main/res/layout/andes_layout_feedbackinapp.xml [hunk 36]
```diff
@@ -5,52 +5,53 @@
...
+            app:layout_constraintTop_toTopOf="@id/andes_feedbackinapp_divider" />
 
         <com.mercadolibre.android.andesui.button.AndesButton
-            android:id="@+id/andes_feedback_button"
+            android:id="@+id/andes_feedbackinapp_button"
             android:layout_width="match_parent"
-            android:layout_height="@dimen/andes_feedback_button_height"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="@dimen/andes_feedback_layout_marginTop"
             android:layout_marginBottom="@dimen/andes_feedback_marginBottom"
```
Comment (mdobratinich) [inline components/src/main/res/layout/andes_layout_feedbackinapp.xml]: igual que el comentario anterior

File: components/src/main/res/layout/andes_layout_feedbackinapp.xml [hunk 37]
```diff
@@ -5,52 +5,53 @@
...
             android:gravity="center_horizontal"
             app:andesButtonHierarchy="transparent"
             app:andesButtonSize="medium"
             app:layout_constraintBottom_toBottomOf="parent"
             app:layout_constraintEnd_toEndOf="parent"
             app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toBottomOf="@+id/andes_descriptionText" />
+            app:layout_constraintTop_toBottomOf="@+id/andes_feedbackinapp_cardText" />
```
Comment (mdobratinich) [inline components/src/main/res/layout/andes_layout_feedbackinapp.xml:54]: igual que el comentario anterior

File: demoapp/build.gradle [hunk 38]
```diff
@@ -105,6 +105,7 @@ dependencies {
     implementation showcase.glide
     implementation showcase.markwon
     debugImplementation showcase.leakCanary
+    implementation 'com.android.support:support-v4:28.0.0'
```
Comment (mauriciogigena) [inline demoapp/build.gradle:108]: esto no deberia estar, salvo que se haya agregado con un pull de develop.

**Review by mauriciogigena — CHANGES_REQUESTED** [review]

File: components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt [hunk 39]
```diff
@@ -0,0 +1,219 @@
...
+    private fun setupFeedbackInAppAlign(config: FeedbackInAppConfiguration) {
+        if (config.cardTextAlignment == AndesFeedbackInAppAlign.LEFT) {
+                    feedbackinappMessage.gravity = Gravity.LEFT
+        }
+        if (config.cardTextAlignment == AndesFeedbackInAppAlign.CENTER) {
+            feedbackinappMessage.gravity = Gravity.CENTER
+        }
+        if (config.cardTextAlignment == AndesFeedbackInAppAlign.RIGHT) {
+            feedbackinappMessage.gravity = Gravity.RIGHT
+        }
```
Comment (mdobratinich) [inline components/src/main/java/com/mercadolibre/android/andesui/feedbackinapp/AndesFeedbackInApp.kt:215]: Cambia la logica de 3 if por un when
```suggestion
        when(config.cardTextAlignment) {
               AndesFeedbackInAppAlign.LEFT -> feedbackinappMessage.gravity = Gravity.LEFT
               AndesFeedbackInAppAlign. RIGHT -> feedbackinappMessage.gravity = Gravity. RIGHT
               else -> feedbackinappMessage.gravity = Gravity.CENTER
        }
```

File: gradle/wrapper/gradle-wrapper.properties [hunk 40]
```diff
@@ -1,5 +1,6 @@
+#Wed Dec 30 09:36:51 CST 2020
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-bin.zip
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-all.zip
```
Comment (mdobratinich) [inline gradle/wrapper/gradle-wrapper.properties:6]: Estos cambios no creo que sean necesarios.. revisalo, sino reverti estos cambios

**Pull request closed without merging**